### ISSUE TITLE
Available for Apex 4.2.x

### ISSUE THREAD

**Issue by hairpin666** [Dani3lSun/apex-plugin-dropzone#4]
Hi,

Is it possible to use the plugin wiht Apex 4.2.x as well? 

Thanks

**Comment by Dani3lSun** [comment]
Hi @hairpin666 
unfortunately this is not possible, a plugin that is developed with APEX 5 is not backward compatible to a 4.2.x release. Therefore I have to create the whole plugin for this release completely again...
I don´t even have a 4.2.x instance which I can use...

Sorry for that, best is to upgrade your instances!;)

**Comment by Dani3lSun** [comment]
Hi @hairpin666 ,
I created a version for APEX 4.2, you can find it here:
https://github.com/Dani3lSun/apex-plugin-dropzone/releases/tag/v1.9.2
Note: Thus APEX 4.2 doesn´t support folders in the plugin files section you have to upload the (js/css/img) folder to your webserver! A note.txt is included...

**Comment by hairpin666** [comment]
Thanks, I gonna test it and give feedback!

**Comment by hairpin666** [comment]
Hi,
did some quick testing, everything works except if I want to upload files which are bigger than 1MB, even thou the limit is set high enough. Is that a bug or do I have to consider sth. else?

Thanks

**Comment by Dani3lSun** [comment]
Hi @hairpin666 
I tested it just now on my APEX 4.2 instance with theme 25/26 and everything works as expected...
The max filesize parameter works for every value I put into this (tested from 1 to 5 MB) and all files get uploaded except the ones that are bigger than the max. allowed size...Tested with ORDS 2.0.10 and Webtier OHS 11g as webserver...Are all plugin files right in place? Any messages in console log? For better debugging you can set the logging param of the plugin to true, after that you see all values logged to console...
<img width="918" alt="bildschirmfoto 2016-02-17 um 20 41 53" src="https://cloud.githubusercontent.com/assets/698942/13122414/0fa4b8ec-d5b7-11e5-80b1-b748c0713cf6.png">

**Comment by hairpin666** [comment]
Hi,

There are no error messages in the console everything looks fine. I even added some console.log calls in the apexdropzone.js to see whats going on. The ajax success function is called after the upload but the file never arrives in the collection or in the custom insert statement. 
We are using Tomcat 7 with ORDS 3.0 on it.

What else could the problem be?

Thanks

**Comment by Dani3lSun** [comment]
Hi,
your first problem was that files bigger than 1MB don´t get uploaded, so the upload was working earlier, except these files? Have you changed anything in the plugin plsql code section? The default code inserts into a collection, if something is missing there, it won´t function well...
If you make changes (console.log entries) in the apexdropzone.js file, it only works if you set your APEX app into debug mode, only then this file is taken, in normal mode the minified version (apexdropzone.min.js) is used...
Can you please test it with a fresh workspace or fresh blank app and of course a not modified plugin file?
Very difficult for me to figure out what the problem could be, on all my instances it works well...Are you using a standard theme or some custom one?

**Comment by Dani3lSun** [comment]
Can you please import this sample app and check if it works?
It only has 1 page with the dropzone plugin. The only thing you have to do is to adjust the PLUGIN_PREFIX path in the dropzone plugin settings under shared components --> plugins
[f100_dropzone_test_apex42.sql.zip](https://github.com/Dani3lSun/apex-plugin-dropzone/files/147115/f100_dropzone_test_apex42.sql.zip)

**Comment by Dani3lSun** [comment]
Anything new on this topic? Does it work? If so I will close the issue...

**Comment by hairpin666** [comment]
Hi,
Sorry for the late reply. 

I tested it again also with your plain application. Same problem here. 

Seems like all files smaller than 5MB get uploaded everything bigger does not get through. 

Could it be any preferences within the workspace / tomcaat / listener?

**Comment by Dani3lSun** [comment]
Hi @hairpin666 
no problem. Just now I could reproduce your issue, because I updated last week to ORDS3+Tomcat.
Large files work with my old settings (ORDS2 standalone + Apache) and OHS11g. But I solved the issue with setting maxPostSize in server.xml of tomcat server (default max size is 2MB)
My tomcat connector looks like this for a maximum of 15MB:
```
<Connector port="8009" protocol="AJP/1.3" redirectPort="8443" address="127.0.0.1" maxPostSize="15728640" />
```
Just add the maxPostSize parameter with a byte value to it and it should work! I will add this hint to the readme file...
